cmdstanpy/model.py

```diff
@@ -84,7 +84,8 @@ class CmdStanModel:
     unless the argument ``compile=False`` is specified.
     The set of constructor arguments are:
 
-    :param model_name: Model name, used for output file names.
+    :param model_name: Deprecated: Model name will be taken from executable.
```

Comment:
Does "Model name will be taken from executable" apply to the current situation or is it what will happen after deprecation?  I'd be inclined to put the actual doc first, then follow with deprecation warning, i.e.,

<tt>
:param model_name: Model name, used for output file names. **Warning**: This argument is deprecated and will be removed in the next release (Version ???). As of Version ???, the model_name will always be taken to be the name of the executable file minus the ``.exe`` suffix if it has one.]
</tt>

Reply:
It describes the default behavior now(if the argument is unspecified) and the only behavior in the next version